stimpacks/aws/login.go

```diff
@@ -1,36 +1,67 @@
 package aws
 
 import (
-	"github.com/skratchdot/open-golang/open"
-
-	"encoding/json"
 	"errors"
 	"fmt"
-	"io/ioutil"
-	"net/http"
-	"net/url"
-	"strings"
 	"time"
+
+	stimaws "github.com/PremiereGlobal/stim/pkg/aws"
+	"github.com/skratchdot/open-golang/open"
 )
 
 // TODO: Move this to a global config
 var stimURL = "https://github.com/PremiereGlobal/stim"
 
+// Profile This is our custom profile we'll use to keep track of things
+type Profile struct {
+	AccessKeyID     string `ini:"aws_access_key_id"`
+	SecretAccessKey string `ini:"aws_secret_access_key"`
+	SessionToken    string `ini:"aws_session_token"`
+	LeaseID         string `ini:"vault_lease_id"`
+}
+
 // Login gets IAM or STS credentials
-// TODO: Update ~/.aws/ config files for cross shell aws acess
 func (a *Aws) Login() error {
+
+	// Create an unauthenticated Aws instance
+	a.aws = a.stim.Aws("", "")
+
 	// Create a Vault instance
 	a.vault = a.stim.Vault()
 
+	// Prompt the user (or get from arguments) the account and role
 	account, role, err := a.GetCredentials()
 	if err != nil {
 		return err
 	}
 	a.log.Debug("Account: ", account, " Role: ", role)
 
+	// Looked for a saved profile if desired
+	useProfiles := a.stim.GetConfigBool("aws.use-profiles")
+	profileName := account + "/" + role
+	if useProfiles {
+		a.log.Debug("Using AWS profiles")
+
+		// Check if we have a saved profile for the right account/role
+		profile := Profile{}
+		err := a.aws.MapProfile(profileName, &profile)
+		if err != nil {
+			return err
+		}
+
+		// Validate we got back the expected fields.  If these are missing, the
+		// profile is missing or invalid so we'll just generate a new one
+		if profile.AccessKeyID != "" && profile.SecretAccessKey != "" && profile.LeaseID != "" {
+			a.log.Debug("Profile " + profileName + " found, validating...")
```

Comment:
more of an fyi, the logger allows you to do string formatting, this will keep the program from having to concat and waste resources for a log message that would never be used:

```golang
  a.log.Debug("Profile {} found, validating...",profileName)
```

Reply:
Fixed all the ones I added